client-v2/src/actions/ArticleFragments.ts

```diff
@@ -235,40 +237,46 @@ const insertArticleFragmentWithCreate = (
   persistTo: 'collection' | 'clipboard',
   // allow the factory to be injected for testing
   articleFragmentFactory = createArticleEntitiesFromDrop
-): ThunkResult<void> => {
-  return (dispatch: Dispatch, getState) => {
-    const insertActionCreator = getInsertionActionCreatorFromType(
-      to.type,
-      persistTo
-    );
-    if (!insertActionCreator) {
-      return;
-    }
-    const sharedState = selectSharedState(getState());
-    const toWithRespectToState = getToGroupIndicesWithRespectToState(
-      to,
-      sharedState,
-      false
-    );
-    if (toWithRespectToState) {
-      return dispatch(articleFragmentFactory(drop))
-        .then(fragment => {
-          if (fragment) {
-            dispatch(
-              insertActionCreator(
-                toWithRespectToState.id,
-                toWithRespectToState.index,
-                fragment.uuid,
-                null
-              )
-            );
-          }
-        })
-        .catch(() => {
-          // @todo: implement once error handling is done
-        });
+): ThunkResult<void> => async (dispatch: Dispatch, getState) => {
+  const insertActionCreator = getInsertionActionCreatorFromType(
+    to.type,
+    persistTo
+  );
+  if (!insertActionCreator) {
+    return;
+  }
+  const sharedState = selectSharedState(getState());
+  const toWithRespectToState = getToGroupIndicesWithRespectToState(
+    to,
+    sharedState,
+    false
+  );
+  if (toWithRespectToState) {
```

Comment:
We discussed this IRL -- this relates to managing articles with orphaned groups. It's not super clear what this means in the code, and this would be a worthy candidate for a refactor, although it probably belongs in a separate PR 👍